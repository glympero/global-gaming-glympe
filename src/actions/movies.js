//SET_MOVIES
const setMovie = (movies) => ({
    type: 'SET_MOVIES',
    movies
  });

// REMOVE_MOVIE
const removeMovie = ({ id } = {}) => ({
    type: 'REMOVE_MOVIE',
    id
});

// EDIT_MOVIE
const editMovie = (id, updates) => ({
    type: 'EDIT_MOVIE',
    id,
    updates
});

export { setMovie, removeMovie, editMovie };