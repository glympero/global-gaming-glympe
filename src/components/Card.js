import React from 'react';
import { Link } from 'react-router-dom';

class Clock extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            day: '',
            date: '',
            month: '',
            year: '',
            hours: '',
            minutes: '',
            sec: '',
            isMounted: false
        }
        let refreshIntervalId;
    }

    getDay = (d) => {
        let day = d.getDay();
        let dayarr =["Sun","Mon","Tues","Wed","Thurs","Fri","Sat"];
        day = dayarr[day];
        return day;
    }

    getDate = (d) => {
        return d.getDate();
    }

    getMonth = (d) => {
        let month = d.getMonth();
        let montharr =["Jan","Feb","Mar","April","May","June","July","Aug","Sep","Oct","Nov","Dec"];
        month = montharr[month];
        return month;
    }

    getYear = (d) => {
        return  d.getFullYear();
    }

    getHours = (d) => {
        return d.getHours();
    }

    getMinutes = (d) => {
        if(d.getMinutes() < 10){
            return `0${d.getMinutes()}`;
        }
        return `${d.getMinutes()}`;
    }

    getSeconds = (d) => {
        if(d.getSeconds() < 10){
            return `0${d.getSeconds()}`;
        }
        return d.getSeconds();
    }
    clock = () => {
        const d = new Date();
        this.setState(() => ({
            day: this.getDay(d),
            date: this.getDate(d),
            month: this.getMonth(d),
            year: this.getYear(d),
            hours: this.getHours(d),
            minutes: this.getMinutes(d),
            sec: this.getSeconds(d)
        }));
      }

      componentDidMount =() => {
        this.refreshIntervalId = setInterval(this.clock, 1000);
      }

    componentWillUnmount = () => {
        clearInterval(this.refreshIntervalId);
    }

    render() {
        return (
            <div className="blog-card spring-fever">
                <Link to='/movies'>
                    <div className="title-content">
                        <h2>Movies List</h2>
                        <hr />
                        <div className="intro">Click Here For Movies List</div>
                    </div>
                    <div className="card-info">Show Movies</div>
                    <div className="utility-info">
                        <ul className="utility-list">
                            <li className="date">{this.state.day} {this.state.date} {this.state.month} {this.state.year}</li>
                            {this.state.hours && <li className="comments">{this.state.hours} : {this.state.minutes} : {this.state.sec}</li>}
                        </ul>
                    </div>
                    <div className="gradient-overlay"></div>
                     <div className="color-overlay"></div>
                </Link>
            </div>
            
        );
    }
};

export default Clock;