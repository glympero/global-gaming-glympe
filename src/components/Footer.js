import React from 'react';

const Footer = () => (
    <div className="footer">
      <div id="button"></div>
      <div id="container">
        <div id="cont">
          <div className="footer_center">
            <div className="footer-social-icons">
              <ul className="social-icons">
                  <li><a href="http://glympero-portfolio.herokuapp.com/" target="_blank" rel="noopener noreferrer" className="social-icon"> <i className="fa fa-globe"></i></a></li>
                  <li><a href="https://twitter.com/lympero" target="_blank" rel="noopener noreferrer" className="social-icon"> <i className="fa fa-twitter"></i></a></li>
                  <li><a href="https://github.com/glympero" target="_blank" rel="noopener noreferrer" className="social-icon"> <i className="fa fa-github"></i></a></li>
                  <li><a href="https://www.linkedin.com/in/george-lymperopoulos-398a6551" target="_blank" rel="noopener noreferrer" className="social-icon"> <i className="fa fa-linkedin"></i></a></li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
);

export default Footer;
