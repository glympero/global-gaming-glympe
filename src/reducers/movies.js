const moviesReducerDefaultState = [];

const moviesReducer = (state = moviesReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_MOVIES':
            return action.movies;
        case 'REMOVE_MOVIE':
            return state.filter(movie => movie.id !== action.id);
        case 'EDIT_MOVIE':
            return state.map(movie => {
                if(movie.id === action.id) {
                    return {
                        ...movie,
                        ...action.updates
                    }
                }
                return movie;
            })
        default:
            return state;
    }
};

export default moviesReducer;