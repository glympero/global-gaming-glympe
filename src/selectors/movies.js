const getVisibleMovies = (movies, filters) => {
    const {text, sortBy, genre } = filters;
    return movies.filter(movie => {
        const textMatched = typeof text === 'string' && movie.title.toLowerCase().indexOf(text.toLowerCase()) !== -1;
        const genreMatched = typeof genre === 'string' && genre === 'all' ? true : movie.genre.toLowerCase().indexOf(genre.toLowerCase()) !== -1;
        return textMatched && genreMatched;
    }).sort((a, b) => {
        if(sortBy === 'title') {
            return a.title > b.title ? 1 : -1;
        }else if (sortBy === 'director') {
            return a.director.toLowerCase() > b.director.toLowerCase() ? 1 : -1;
        }
    });
};

export default getVisibleMovies;