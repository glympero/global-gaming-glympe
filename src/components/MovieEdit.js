import React from 'react';
import PropTypes from 'prop-types';
import MovieModal from './MovieModal';
import { connect } from 'react-redux';
import { editMovie, removeMovie } from '../actions/movies';

export class MovieEdit extends React.Component {
  
  onRemove = () => {
    this.props.removeMovie( {id: this.props.movie.id} );
    this.props.onClose();
  }

  onSubmit = (movie) => {
    this.props.editMovie(this.props.movie.id, movie);
    this.props.onClose();
  }

  render() {
    return (
        <MovieModal
          onSubmit={this.onSubmit}
          movie={this.props.movie}
          remove={this.onRemove}
        />
    )
  }
}

const mapStateToProps = (state, props) => {
  let movie = state.movies.find( e => e.id === props.id);
  return {
    movie
  }
};

const mapDispatchToProps = (dispatch, props) => {
  return {
    removeMovie: (data) => {
      dispatch(removeMovie(data));
    },
    editMovie: (id, movie) => {
      dispatch(editMovie(id, movie));
    }
  }
};

MovieEdit.propTypes = {
  removeMovie: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired,
  editMovie: PropTypes.func.isRequired,
  movie: PropTypes.object
};

export default connect(mapStateToProps, mapDispatchToProps)(MovieEdit);