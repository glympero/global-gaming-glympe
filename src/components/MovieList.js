import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { setMovie } from '../actions/movies';
import MovieListItem from './MovieListItem';
import selectmovies from '../selectors/movies';

import data from '../fixtures/movies';

export class MovieList extends React.Component {

  componentDidMount = () => {
    this.props.setMovie(JSON.parse(data));
  }

  render() {
    return (
      <div className='content-container'>
        <div className='list-body'>
          {
            this.props.movies.length === 0 ? (
              <p>No movies Found</p>
            ) : (
              this.props.movies.map((movie) => {
                return (
                  <MovieListItem key={movie.id}  {...movie} />
                );
              })
            )
          }
        </div>
      </div>    
    );
  }
};

const mapStateToProps = (state) => {
  return {
    movies: selectmovies(state.movies, state.filters)
  }
};

const mapDispatchToProps = (dispatch, props) => {
  return {
    setMovie: (data) => {
      dispatch(setMovie(data));
    }
  }
};

MovieList.propTypes = {
  movies: PropTypes.array,
  setMovie: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(MovieList);
