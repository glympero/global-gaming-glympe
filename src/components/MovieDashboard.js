import React from 'react';
import Title from './Title';
import MovieList from './MovieList';
import MovieListFilters from './MovieListFilters';
import Card from './Card';

const MovieDashboard = (props) => {

    return (
      <div>
        <Title title="Welcome to our Movies database" />
        <Card />
    </div>
    );
  };

export default MovieDashboard;