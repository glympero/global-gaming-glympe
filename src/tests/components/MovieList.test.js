import React from 'react';
import { shallow } from 'enzyme';

import { MovieList } from '../../components/MovieList'; 
import movies from '../../fixtures/data';

let dispatch, setMovie
beforeEach(() => {
    setMovie = jest.fn();
    dispatch = jest.fn();
});


test('should render MovieList with movies', () => {
    const wrapper = shallow(<MovieList movies={movies} dispatch={dispatch} setMovie={setMovie}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render MovieList without users', () => {
    const wrapper = shallow(<MovieList movies={[]} dispatch={dispatch} setMovie={setMovie}/>)
    expect(wrapper).toMatchSnapshot();
});