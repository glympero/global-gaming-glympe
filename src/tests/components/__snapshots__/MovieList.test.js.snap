// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieList with movies 1`] = `
<div
  className="content-container"
>
  <div
    className="list-body"
  >
    <MovieListItem
      director={
        Object {
          "gender": "Female",
          "name": "Walsh Mertie",
        }
      }
      genre="Drama@april.biz"
      id="1"
      key="1"
      poster_path="12345"
      release_date="12 June 2017"
      title="Absolute Bite"
    />
    <MovieListItem
      director={
        Object {
          "gender": "Male",
          "name": "Grady Macey",
        }
      }
      genre="Children"
      id="2"
      key="2"
      poster_path="12345"
      release_date="26 June 2017"
      title="Alarmed Department"
    />
    <MovieListItem
      director={
        Object {
          "gender": "Female",
          "name": "McKenzie Hertha",
        }
      }
      genre="Mystery"
      id="3"
      key="3"
      poster_path="12345"
      release_date="18 December 2017"
      title="Tight Ticket"
    />
  </div>
</div>
`;

exports[`should render MovieList without users 1`] = `
<div
  className="content-container"
>
  <div
    className="list-body"
  >
    <p>
      No movies Found
    </p>
  </div>
</div>
`;
